Backends/CLX/medium.lisp

```diff
@@ -124,19 +140,36 @@
         (unless (eq (line-style-joint-shape line-style)
                     (line-style-joint-shape old-line-style))
           (setf (xlib:gcontext-join-style gc)
-                (line-style-joint-shape line-style)))
+                (translate-join-shape (line-style-joint-shape line-style))))
         ;; we could do better here by comparing elements of the vector
         ;; -RS 2001-08-24
         (unless (and (eq (line-style-dashes line-style)
                          (line-style-dashes old-line-style))
                      (eq (line-style-unit line-style)
                          (line-style-unit old-line-style)))
-          (setf (xlib:gcontext-line-style gc)
-                (if (line-style-dashes line-style) :dash :solid)
-                (xlib:gcontext-dashes gc)
-                (case (line-style-dashes line-style)
-                  ((t nil) (round (* (line-style-scale line-style medium) 3)))
-                  (otherwise (line-style-effective-dashes line-style medium)))))))))
+          (let ((effective (line-style-effective-dashes line-style medium)))
+            (if-let ((dashes (line-style-dashes line-style)))
+              (setf (xlib:gcontext-line-style gc) :dash
+                    (xlib:gcontext-dashes gc) effective)
+              (setf (xlib:gcontext-line-style gc) :solid
+                    (xlib:gcontext-dashes gc) effective))))))))
+
+
+(defmethod (setf medium-transformation) :around (transformation (medium clx-medium))
+  (let ((old-tr     (medium-transformation medium))
+        (line-style (medium-line-style medium))
+        (new-tr     (call-next-method)))
+    (with-slots (gc) medium
+      (when (and gc
+                 (eq :coordinate (line-style-unit line-style))
```

Comment:
If `:coordinate` is not the default, maybe do this check first and skip everything else if possible.

Reply:
we need to call the next method anyway, so the first let is justified, I've made it easier for the eye with

```
(defmethod (setf medium-transformation) :around (transformation (medium clx-medium))
  (let ((old-tr     (medium-transformation medium))
        (line-style (medium-line-style medium))
        (new-tr     (call-next-method)))
    (unless (and (eq :coordinate (line-style-unit line-style))
                 (not (transformation-equal old-tr new-tr)))
      (when-let ((gc (slot-value medium 'gc)))
        (let ((thickness    (line-style-effective-thickness line-style medium))
              (dash-pattern (line-style-effective-dashes    line-style medium))
              (dashesp      (if (line-style-dashes line-style) :dash :solid)))
          (setf (xlib:gcontext-line-style gc) dashesp
                (xlib:gcontext-dashes gc)     dash-pattern
                (xlib:gcontext-line-width gc) thickness))))
    new-tr))
```